app/views/campaigns/_course_row.html.haml

```diff
@@ -57,5 +57,18 @@
       %a.course-link{:href => "#{course_slug_path(course.slug)}"}
         = form_for(@campaign, url: remove_course_campaign_path(@campaign.slug, course_id: course.id), method: :put, id: "remove_course-#{course.id}", html: { class: 'remove-program-form' }) do
           = hidden_field_tag('course_title', course.title, id: "course_title-#{course.id}")
-          %button.button.danger.remove-course{'data-id' => course.id, 'data-title' => course.title, 'data-campaign-title' => @campaign.title}
+          %button.button.danger.remove-course{ 'data-id' => course.id, 'data-title' => course.title, 'data-campaign-title' => @campaign.title, title: t('campaign.remove_course_tooltip') }
             = t('assignments.remove')
+    - if @presenter&.can_delete_course?
+      - if course.campaigns.size > 1
```

Comment:
I would like to avoid needing to query every course for the number of campaigns, as this creates an "N+1" query with an additional database query for every course on the page. (If we can know it can't be deleted already at the time of rendering, then we shouldn't show the delete button.)

I think the right way to do this is to render the same button format for each course, and just detect whether or not it can be deleted during the request to the deletion controller.